packages/@ourworldindata/grapher/src/dataTable/DataTable.tsx

```diff
@@ -657,15 +684,26 @@ export class DataTable extends React.Component<{
         )
     }
 
-    @computed private get entityNames(): string[] {
-        const tableForEntities = this.table
+    @computed private get availableEntityNames(): EntityName[] {
         return union(
             ...this.columnsToShow.map(
-                (col) => tableForEntities.get(col.slug).uniqEntityNames
+                (col) => this.table.get(col.slug).uniqEntityNames
             )
         )
     }
 
+    @computed get fuzzy(): FuzzySearch<string> {
+        return FuzzySearch.withKey(
+            this.availableEntityNames,
+            (entityName) => entityName
+        )
+    }
+
+    @computed private get entityNames(): EntityName[] {
+        if (!this.tableConfig.search) return this.availableEntityNames
+        return this.fuzzy.search(this.tableConfig.search)
```

Comment:
Did you investigate different threshold options? It's maximally permissive at the moment, which might be fine, but maybe it could be more user friendly with a little pruning.

Reply:
I haven't but we use the same fuzzy search in other places and it has never bothered me. Do you have something particular in mind?